### ISSUE TITLE
Syslog driver: TCP keepalive not working

### ISSUE THREAD

**Issue by vostrelt** [syslog-ng/syslog-ng#4546]
# syslog-ng
## Version of syslog-ng
syslog-ng 4 (4.0.0)
Config version: 4.0
Installer-Version: 4.0.0
Revision: 
Compile-Date: Jul 12 2023 10:33:30
Module-Directory: /lib/syslog-ng
Module-Path: /lib/syslog-ng
Include-Path: /share/syslog-ng/include
Available-Modules: system-source,syslogformat,linux-kmsg-format,affile,afsocket,afprog,afuser,basicfuncs
Enable-Debug: off
Enable-GProf: off
Enable-Memtrace: off
Enable-IPv6: off
Enable-Spoof-Source: off
Enable-TCP-Wrapper: off
Enable-Linux-Caps: off
Enable-Systemd: off

## Platform
Third-party Linux, version 4.9.79, MIPS64 architecture.

# Issue
## Failure
I use `syslog` driver to send logs to remote station. I can see TCP keepalive messages are sent if I use `syslog-ng` version 3.30.1 and I can set their time, interval and probes or disable them. I wanted to upgrade to `syslog-ng` version 4.0.0 (my linux kernel does not support SO_MEMINFO, so I cannot use 4.2.0) but the TCP keepalive messages are not sent in this version (not in 3.38.1 or 4.1.1 as well).

## Configuration
```
syslog("192.168.169.13"
		port(514)
		transport(tcp)
		keep-alive(no)
		log-fifo-size(32)
		flush-lines(1)
		suppress(5)
		throttle(2)
		so-keepalive(yes)
		tcp-keepalive-probes(6)
		tcp-keepalive-intvl(30)
		tcp-keepalive-time(300)
	);
```

**Comment by vostrelt** [comment]
It seems like it has something to do with moving the following line in file `modules/afsocket/socket-options-inet.c`.
```
socket_options_inet_setup_tcp_keepalive_timers(self, fd);
```
It was done in commit 3891ace1f afsocket: move TCP keepalive options setting after `accept()` call.
If I copy the line to previous location, the TCP keepalives are sent.

**Comment by MrAnno** [comment]
Thanks for opening this issue.

> my linux kernel does not support SO_MEMINFO, so I cannot use 4.2.0

That's a bug, #4548 will fix it. Sorry for the inconvenience.

I will soon take a look at the keep-alive issue as well.

**Comment by vostrelt** [comment]
Thanks for quick response and opening the issue #4548.

What do you think the next release date is?

**Comment by MrAnno** [comment]
@vostrelt I believe reverting the corresponding PR or the mentioned commit is safe on Linux.

syslog-ng 4.3 is expected at the end of next week. Hopefully we can identify the problem until then.

**Comment by MrAnno** [comment]
Thanks again for the bug report.

The destination side keep-alive is, indeed, broken (caused by #3751). I'm working on a fix.

**Comment by vostrelt** [comment]
> I'm working on a fix.

Great, thanks.

**Comment by MrAnno** [comment]
Both the `SO_MEMINFO` and the TCP keepalive fixes are included in syslog-ng v4.3.0.

**Comment by vostrelt** [comment]
Thank you, I did basic testing and it looks good! :+1:
